core/src/main/java/com/amplifyframework/auth/AuthSignedInStatus.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth;
+
+public enum AuthSignedInStatus {
+    /** User is signed in. **/
+    SIGNED_IN,
+    /** User is signed out. **/
+    SIGNED_OUT,
+    /** User is signed in as a guest. **/
```

Comment:
Or sign in has not been attempted?

Reply:
No if sign in hasn't been attempted they'll always be in SIGNED_OUT state - you have to explicitly sign in as guest.